src/dictionaries.cpp

```diff
@@ -61,24 +62,44 @@ Dictionaries::Dictionaries(const std::string& filePath)
         {
             if (tokens[0] == "ATTRIBUTE")
             {
-                std::string name = tokens[1];
-                std::string code = tokens[2];
-                if (vendorFlag)
-                    std::cout << "  " << name << ": " << code << "\n";
+                const auto& attrName = tokens[1];
+                const auto& code = std::stoul(tokens[2]);
+                if (vendorName.size() != 0)
```

Comment:
Instead of checking string (or any other container) size against zero you better use `empty()` method. In some cases `size()` has linear complexity, when `empty()` is always guaranteed to have constant complexity. Your condition would be:
```c++
if (!vendorName.empty())
```
Also, it is semantically more correct.